test/metabase/query_processor_test/aggregation_test.clj

```diff
@@ -284,6 +284,16 @@
                (or (-> results mt/cols first)
                    results)))))))
 
+(deftest semantic-type-for-aggregate-fields-test
```

Comment:
Might want to put a ticket number for https://github.com/metabase/metabase/issues/38022 in the test name or the testing string.